deepcell_toolbox/metrics.py

```diff
@@ -226,6 +233,22 @@ def __init__(self,
         }
         self.catastrophe_indices['y_pred'] = []
 
+        # If 2D, dimensions can be 2 or 3 (with or without channel dimension)
+        if not self.is_3d:
+            if self.y_true.ndim not in {2, 3}:
+                raise ValueError('Expected dimensions for y_true (2D data) are 2, 3 or 4.'
+                                 'Accepts: (x, y), or (x, y, chan)'
+                                 'Got ndim: {}'.format(self.y_true.ndim))
+
+        # If 3D, inputs must have 3 dimensions (batch, z, x, y) - cannot have channel dimension or
+        # _classify_graph breaks, as it expects input to be 2D or 3D
+        # TODO - add compatibility for multi-channel 3D-data
+        else:
+            if self.y_true.ndim != 3:
+                raise ValueError('Expected dimensions for y_true (3D data) is 3.'
+                                 'Requires format is: (z, x, y)'
+                                 'Got ndim: {}'.format(self.y_true.ndim))
```

Comment:
Does `_classify_graph` support multi-channel 2D data? If so, what are the issues with supporting 3D multi-channel?

Reply:
Yeah it supports multi-channel 2D data. The issue is that regionprops only supports 2D and 3D data, but a multichannel 3D image is effectively 4D.